### PR TITLE
Minor sceNpTrophy improvements, Idm assert fix

### PR DESCRIPTION
**Pull request by elad335**
* Fix an off by one "bug" in idm traits static asserts.
* Fix recursive locking of shared lock in scepTrophyUnlockTrophy.
* Log returned required space in sceNpTrophyGetRequiredDiskSpace, and add logging for other corner cases.

### CODE REVIEW

File: rpcs3/Emu/IdManager.cpp [hunk 1]
```diff
@@ -19,13 +19,9 @@ id_manager::id_map::pointer idm::allocate_id(const id_manager::id_key& info, u32
...
 		// Try to emplace back
 		const u32 _next = base + step * ::size32(vec);
-
-		if (_next >= base && _next < base + step * count)
-		{
-			g_id = _next;
-			vec.emplace_back(id_manager::id_key(_next, info.type()), nullptr);
-			return &vec.back();
-		}
+		g_id = _next;
```
Comment (VelocityRa) [inline rpcs3/Emu/IdManager.cpp]: Change not mentioned in commit name/desc.

Comment (elad335) [reply rpcs3/Emu/IdManager.cpp]: Dead condition anyways.

**Comment by uaqlover** [comment]
also LittleBigPlanet [BCUS98148] have same issue

**Comment by uaqlover** [comment]
asking only could fix https://github.com/RPCS3/rpcs3/issues/5948 it show sceNpTrophyRegister ?

**Comment by elad335** [comment]
Reduced the pr for now, the previosuly rferenced issue needs fixing possibly from ppu instructions code and not sceNpTrophy.
Also added a regression fix from #7111 because show_trophy_notification could have locked trophy mutex while trophy manager mutex was already locked.

**Comment by elad335** [comment]
Improved recursive locking fix.

File: rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp [hunk 2]
```diff
@@ -1070,6 +1047,33 @@ error_code sceNpTrophyGetTrophyInfo(u32 context, u32 handle, s32 trophyId, vm::p
...
+	sceNpTrophy.warning("sceNpTrophyGetTrophyInfo(context=0x%x, handle=0x%x, trophyId=%d, details=*0x%x, data=*0x%x)", context, handle, trophyId, details, data);
+
+	if (trophyId < 0 || trophyId > 127) // max 128 trophies
+	{
+		return SCE_NP_TROPHY_ERROR_INVALID_TROPHY_ID;
+	}
+
+	const auto trophy_manager = g_fxo->get<sce_np_trophy_manager>();
+
+	std::shared_lock lock(trophy_manager->mtx);
```
Comment (Nekotekina) [inline rpcs3/Emu/Cell/Modules/sceNpTrophy.cpp]: Shared lock doesn't allow recursion though. I think you forgot to unlock it manually.

**Pull request merged**